client/utils/account-fees.tsx

```diff
@@ -85,19 +85,39 @@ export const getCurrentBaseFee = (
 		: accountFees.base;
 };
 
+// If the current fee doesn't have a fixed or percentage rate, use the base fee's rate. Eg. when there is a promotional discount fee applied.
+const getCurrentFeeWithBaseFallback = (
+	currentFee: BaseFee | DiscountFee,
+	baseFee: BaseFee
+) => {
+	if ( ! currentFee.percentage_rate ) {
+		currentFee.percentage_rate = baseFee.percentage_rate;
+	}
+
+	if ( ! currentFee.fixed_rate ) {
+		currentFee.fixed_rate = baseFee.fixed_rate;
+	}
+
+	return currentFee;
+};
+
 export const formatMethodFeesTooltip = (
 	accountFees: FeeStructure
 ): JSX.Element => {
 	if ( ! accountFees ) return <></>;
 	const currentBaseFee = getCurrentBaseFee( accountFees );
+	const currentFeeWithBaseFallBack = getCurrentFeeWithBaseFallback(
```

Comment:
Thanks Nagesh!

Apologies for being so annoying, but I don't like very much the fact that we have `currentBaseFee` and `currentFeeWithBaseFallBack`. I still find it does not help with clarifying things. Maybe it is just a matter of adding more comments or simply personal preference. I'd like to hear other opinions.

In case it helps, I'd go with:

```diff
	index b2b9151a..72dcf3f7 100644
--- a/client/utils/account-fees.tsx
+++ b/client/utils/account-fees.tsx
@@ -106,18 +106,14 @@ export const formatMethodFeesTooltip = (
 ): JSX.Element => {
        if ( ! accountFees ) return <></>;
        const currentBaseFee = getCurrentBaseFee( accountFees );
-       const currentFeeWithBaseFallBack = getCurrentFeeWithBaseFallback(
-               currentBaseFee,
-               accountFees.base
-       );
 
        const total = {
                percentage_rate:
-                       currentFeeWithBaseFallBack.percentage_rate +
+                       currentBaseFee.percentage_rate +
                        accountFees.additional.percentage_rate +
                        accountFees.fx.percentage_rate,
                fixed_rate:
-                       currentFeeWithBaseFallBack.fixed_rate +
+                       currentBaseFee.fixed_rate +
                        accountFees.additional.fixed_rate +
                        accountFees.fx.fixed_rate,
                currency: accountFees.base.currency,
@@ -132,7 +128,12 @@ export const formatMethodFeesTooltip = (
                        <div>
                                <div>Base fee</div>
                                <div>
-                                       { getFeeDescriptionString( currentFeeWithBaseFallBack ) }
+                                       { getFeeDescriptionString(
+                                               getCurrentFeeWithBaseFallback(
+                                                       currentBaseFee,
+                                                       accountFees.base
+                                               )
+                                       ) }
                                </div>
                        </div>
                        { hasFees( accountFees.additional ) ? (
@@ -156,7 +157,9 @@ export const formatMethodFeesTooltip = (
                        <div>
                                <div>Total per transaction</div>
                                <div className={ 'wcpay-fees-tooltip__bold' }>
-                                       { getFeeDescriptionString( total ) }
+                                       { getFeeDescriptionString(
+                                               getCurrentFeeWithBaseFallback( total, accountFees.base )
+                                       ) }
                                </div>
                        </div>
                        { wcpaySettings &&

```

But, as I said, this may be just a personal preference as both approaches do the same.

Reply:
>Apologies for being so annoying, but I don't like very much the fact that we have currentBaseFee and currentFeeWithBaseFallBack

No need to apologize, and this is not annoying at all. Infact your opinion and stress on making the code easily understandable makes a lot of sense. Let me go back to the drawing board and test out some ideas I see here.

Reply:
Things are slightly tricky here because of the `total` variable and how we need to add the fallback base rate, component-wise - for `percentage` and `fix rates`.  @mgascam , the last suggestion ends up showing the base fee as the total fee, without adding the additional and FX fees to it. 

<img width="347" alt="image" src="https://user-images.githubusercontent.com/4162931/212464727-b5815fc4-0e24-465e-9273-02aaf804ae2a.png">

This happens because the fallback rates, if applicable, need to be added separately for `percentage` and `fixed` rate, where we [first compute](https://github.com/Automattic/woocommerce-payments/blob/ae30b742c5a0880919f363cfabc64dbd6b4e4c1c/client/utils/account-fees.tsx#L109) `total`.  This adjusted rate needs to be then given as an input to the string generation function. That's why I opted to create a separate const earlier to store the current fee with base fall back ( if applicable ) . 

I have added comments in my last update here. Let me know if this makes sense.

Reply:
With reference to my other comment, IMO it'll be enough to do `currentFeeWithBaseFallBack = accountFees.base`.
And fixing the names is `currentBaseFee = accountFees.base`

Reply:
Thanks for the critical review and guidance. I have simplified this and made amends here 
6db016a

I hope this will work fine.

Reply:
@mgascam May I request your review too? Thanks!